homeassistant/components/modbus/sensor.py

```diff
@@ -131,32 +149,35 @@ async def async_setup_platform(
                     "Unable to detect data type for %s sensor, try a custom type",
                     entry[CONF_NAME],
                 )
-                continue
+                return None, None
         else:
-            structure = entry.get(CONF_STRUCTURE)
+            structure = entry.get(CONF_STRUCTURE, "")
```

Comment:
If the user defined data_type: custom, and did not add structure there should be a _LOGGER.warning() 
Yes I know you copied the code, but why not improve it :-)

Reply:
Sure.
As I think about it, it seems it should be a _LOGGER.error() with its own tests. The following try/catch will handle it, but the message will be somewhat misleading. I will do an update shortly

Reply:
We only use error() if the platform fails and are not loaded.....so warnings() are in many cases errors in human eyes.

Reply:
Ok, renamed then all errors to warnings + tests